src/java.base/share/classes/java/lang/reflect/code/type/ClassType.java

```diff
@@ -129,27 +129,18 @@ public JavaType toBasicType() {
     }
 
     public String toClassName() {
-        return type;
+        String pkg = type.packageName();
+        return pkg.isEmpty() ?
+                type.displayName() :
+                String.format("%s.%s", pkg, type.displayName());
     }
 
     public String toInternalName() {
```

Comment:
I've dropped `toInternalName` as it didn't seem used and, if we expose `ClassDesc` that seems enough?